### PR TITLE
Change default and deprecate requireJREPackageImports, add release note

### PR DESCRIPTION
**Pull request by laeubi**
Fixes https://github.com/eclipse/tycho/issues/1136

### CODE REVIEW

**Comment by laeubi** [comment]
At least none of the integration test care here...

**Comment by mickaelistria** [comment]
>   You requested to install 'osgi.bundle; org.bouncycastle.bcpg 0.0.0' but it could not be found

This should be fixed now.
If there is some risk of Tycho being broken, I'd rather wait after next I-Build is complete. We didn't have a good build of Platform for 4 days; having 1 proper one before changing Tycho would be useful.

File: tycho-compiler-plugin/src/main/java/org/eclipse/tycho/compiler/AbstractOsgiCompilerMojo.java [hunk 1]
```diff
@@ -642,7 +657,14 @@ private void configureBootclasspathAccessRules(CompilerConfiguration compilerCon
             throws MojoExecutionException {
         List<AccessRule> accessRules = new ArrayList<>();
 
-        if (requireJREPackageImports) {
+        if (requireJREPackageImports != null) {
```
Comment (mickaelistria) [inline tycho-compiler-plugin/src/main/java/org/eclipse/tycho/compiler/AbstractOsgiCompilerMojo.java:660]: boolean cannot be null, so warning would always show up.

**Review by mickaelistria — CHANGES_REQUESTED** [review]

**Comment by mickaelistria** [comment]
Actually, you don't need to merge this patch to test it on platform. Just submit a PR to the platform parent pom that set the existing flag to `true` and you'll see what happens.

**Comment by laeubi** [comment]
> Actually, you don't need to merge this patch to test it on platform.

I'll run a aggregator local build, I assume everything is fixed now?

**Comment by mickaelistria** [comment]
>  assume everything is fixed now?

Yes.

> I'll run a aggregator local build,

As you prefer, but sending a PR will trigger a full build without running tests on CI.

**Pull request merged**